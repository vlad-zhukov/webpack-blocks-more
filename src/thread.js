import threadLoader from 'thread-loader';

/*
 * Thread loader.
 */
export default function thread(options = {}) {
    const {warmupModules, ...restOptions} = options;
    if (warmupModules) {
        threadLoader.warmup(restOptions, warmupModules);
    }

    return (context, {addLoader}) =>
        addLoader({
            ...context.match,
            use: [
                {
                    loader: 'thread-loader',
                    options: restOptions,
                },
            ],
        });
}
