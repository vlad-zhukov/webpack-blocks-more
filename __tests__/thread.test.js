import {createConfig, match} from '@webpack-blocks/webpack';
import {file, css} from '@webpack-blocks/assets';
import thread from '../src/thread';

it('should work with match()', () => {
    const config = createConfig([match('*.js', [thread()])]);

    expect(config).toMatchSnapshot();
});

it('should take options', () => {
    const config = createConfig([
        match('*.js', [
            thread({
                name: 'my-pool',
            }),
        ]),
    ]);

    expect(config).toMatchSnapshot();
});

it('should work with other blocks', () => {
    const config = createConfig([
        match('*.js', [
            file({
                name: '[hash:20].css',
            }),
            thread(),
            css(),
        ]),
    ]);

    expect(config).toMatchSnapshot();
});
