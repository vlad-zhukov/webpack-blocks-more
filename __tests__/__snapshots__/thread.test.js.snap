// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take options 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\^\\.\\*\\\\\\.js\\$/,
        "use": Array [
          Object {
            "loader": "thread-loader",
            "options": Object {
              "name": "my-pool",
            },
          },
        ],
      },
    ],
  },
  "plugins": Array [],
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".json",
    ],
  },
  "stats": Object {
    "children": false,
    "chunks": false,
    "modules": false,
    "reasons": false,
  },
}
`;

exports[`should work with match() 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\^\\.\\*\\\\\\.js\\$/,
        "use": Array [
          Object {
            "loader": "thread-loader",
            "options": Object {},
          },
        ],
      },
    ],
  },
  "plugins": Array [],
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".json",
    ],
  },
  "stats": Object {
    "children": false,
    "chunks": false,
    "modules": false,
    "reasons": false,
  },
}
`;

exports[`should work with other blocks 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\^\\.\\*\\\\\\.js\\$/,
        "use": Array [
          Object {
            "loader": "file-loader",
            "options": Object {
              "name": "[hash:20].css",
            },
          },
          Object {
            "loader": "thread-loader",
            "options": Object {},
          },
          Object {
            "loader": "style-loader",
            "options": Object {},
          },
          Object {
            "loader": "css-loader",
            "options": Object {},
          },
        ],
      },
    ],
  },
  "plugins": Array [],
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".json",
    ],
  },
  "stats": Object {
    "children": false,
    "chunks": false,
    "modules": false,
    "reasons": false,
  },
}
`;
